lib/src/core.dart

```diff
@@ -307,10 +307,6 @@ class AudioSession {
     } else if (_avAudioSession != null) {
       final currentRoute = await _avAudioSession!.currentRoute;
       if (includeInputs) {
-        final darwinInputs = await _avAudioSession!.availableInputs;
-        devices.addAll(darwinInputs
-            .map((port) => _darwinPort2device(port, inputPorts: darwinInputs))
-            .toSet());
```

Comment:
Rather than move this down after the outputs, wouldn't it work to just swap the order of these two `addAll`s?

Also based on your comment, I think it would actually make sense to pass the `outputPorts:` parameter here, too. And before any of this starts, I should collect all the input ports and all the output ports into two variables which I can then pass in here and on each invocation.